fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseRunner.java

```diff
@@ -432,12 +438,30 @@ public T run(@Nonnull Function<? super FDBRecordContext, ? extends T> retriable)
      *
      * @param retriable the database operation to run transactionally
      * @param <T> return type of function to run
-     * @return future that will contain the result of function after successful run and commit
+     * @return future that will contain the result of {@code retriable} after successful run and commit
      * @see #run(Function)
      */
     @Nonnull
     public <T> CompletableFuture<T> runAsync(@Nonnull Function<? super FDBRecordContext, CompletableFuture<? extends T>> retriable) {
-        return new RunRetriable<T>().runAsync(retriable);
+        return runAsync(retriable, Pair::of);
+    }
+
+    /**
+     * Runs a transactional function asynchronously with retry logic.
+     * This is also a non-blocking call.
+     *
+     * @param retriable the database operation to run transactionally
+     * @param handlePostTransaction after the transaction is committed, or fails to commit, this function is called with the
+     * result or exception respectively. This handler should return a new pair with either the result to return from
```

Comment:
I think the second sentence of this parameter description might fit our parameter style better if it were mentioned in the main body of the Javadoc description.

Reply:
I feel like that would make things worse, considering the second sentence is specifically what the handler should return, and not about the overall behavior of the method.

Reply:
Hm, I don't think we generally have multi-statement parameter descriptions, but fine.